lib/services/livesync/livesync-service.ts

```diff
@@ -34,16 +37,61 @@ export class LiveSyncService implements ILiveSyncService {
 
 			let projectDir = this.$project.getProjectDir().wait();
 
-			let livesyncData = {
+			let livesyncData: ILiveSyncData = {
 				platform: platform,
 				appIdentifier: this.$project.projectData.AppIdentifier,
 				projectFilesPath: projectDir,
 				syncWorkingDirectory: projectDir,
-				excludedProjectDirsAndFiles: this.excludedProjectDirsAndFiles
+				excludedProjectDirsAndFiles: this.excludedProjectDirsAndFiles,
+				additionalConfigurations: this.$project.projectInformation.configurations
 			};
 
+			let deviceConfigurationInfos: Mobile.IApplicationInfo[] = [];
+
+			let configurations = this.$project.getConfigurationsSpecifiedByUser();
+			if (!configurations.length) {
+				this.$devicesService.execute(device => (() => {
+					let configInfo = device.getApplicationInfo(livesyncData.appIdentifier).wait();
+					if (configInfo) {
+						deviceConfigurationInfos.push(configInfo);
+					}
+				}).future<void>()()).wait();
+
+				let deviceConfigurations = _.reduce(deviceConfigurationInfos, (result, dci) => result + `${EOL}device: ${dci.applicationIdentifier} has "${dci.configuration}" configuration`, "");
+				if (deviceConfigurations && _.uniq(deviceConfigurationInfos, dci => dci.configuration).length !== 1) {
+						this.$errors.failWithoutHelp("Cannot LiveSync because application is deployed with different configurations across the devices.", deviceConfigurations);
+				}
+
+				livesyncData.configuration = deviceConfigurationInfos && deviceConfigurationInfos[0] && deviceConfigurationInfos[0].configuration;
+				if (livesyncData.configuration) {
+					this.$options.config = [livesyncData.configuration];
+				}
+
 			$liveSyncServiceBase.sync(livesyncData).wait();
+			} else {
+				configurations.forEach(configuration => {
+					livesyncData.configuration = configuration;
+					livesyncData.appIdentifier = this.$project.projectInformation.configurationSpecificData[configuration.toLowerCase()].AppIdentifier;
```

Comment:
If we want to support case sensitive project configurations we do not need this `.toLowerCase()`